internal/cache/lru/lru.go

```diff
@@ -70,12 +71,14 @@ type entry struct {
 	Value ValueType
 }
 
-// New initialize a cache with the supplied maxSize, which must be greater than
+// Init initialize a cache with the supplied maxSize, which must be greater than
 // zero.
-func NewCache(maxSize uint64) (c Cache) {
+func (c *Cache) Init(maxSize uint64) {
 	c.maxSize = maxSize
 	c.index = make(map[string]*list.Element)
-	return
+
+	// Set up invariant checking.
+	c.mu = locker.New("LRUCache", c.CheckInvariants)
 }
 
 // CheckInvariants panic if any internal invariants have been violated.
```

Comment:
we can make checkInvariants private and remove "The careful user can arrange to call this at crucial moments."?